tests/test_ujson.py

```diff
@@ -600,6 +600,29 @@ def test_decode_numeric_int_exp(test_input):
     assert output == json.loads(test_input)
 
 
+@pytest.mark.parametrize(
+    "i",
+    [
+        -(10**25),  # very negative
+        -(2**64),  # too large in magnitude for a uint64
+        -(2**63) - 1,  # too small for a int64
+        2**64,  # too large for a uint64
+        10**25,  # very positive
+    ],
+)
+@pytest.mark.parametrize("mode", ["encode", "decode"])
+def test_encode_decode_big_int(i, mode):
+    # Test ints that are too large to be represented by a C integer type
+    for py, j in ((i, str(i)), ([i], f"[{i}]"), ({"i": i}, f'{{"i":{i}}}')):
```

Comment:
How about?
```suggestion
    for py in (i, [i], {"i": i}):
        j = json.dumps(py)
```
Those json f-strings are just a bit too ugly.

Reply:
Not quite as `json.dumps` includes spaces after colons in dicts, so `separators` is needed. But otherwise, yeah, that's neater.

Reply:
For the record, Python's own test suite has [decent coverage](https://github.com/python/cpython/blob/f8e576be0a7cd38f753f31cf4178db81a602fc32/Lib/test/test_json/test_enum.py) that `json.dumps` works correctly for these numbers on their own and in lists/dicts.